src/util/locktime.rs

```diff
@@ -0,0 +1,516 @@
+    /// whole benefit of the `LockTime` type (ensuring height/time values are not mixed up).
+    ///
+    /// If you want to compare lock time values consider using `is_satisfied` instead of directly
+    /// comparing the u32's returned by this function.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use bitcoin::LockTime;
+    ///
+    /// let n_lock_time: u32 = 741521;
+    /// let lock_time = LockTime::from(n_lock_time);
+    /// assert_eq!(lock_time.to_u32(), n_lock_time);
+    /// ```
+    pub fn to_u32(&self) -> u32 {
+        match *self {
+            LockTime::Blocks(ref h) => h.to_u32(),
+            LockTime::Seconds(ref t) => t.to_u32(),
+        }
+    }
+}
+
+impl From<u32> for LockTime {
+    /// Constructs a `LockTime` from an nLockTime value or the argument to OP_CHEKCLOCKTIMEVERIFY.
+    fn from(n: u32) -> Self {
+        if is_block_height(n) {
+            Self::Blocks(Height::try_from(n).expect("n is valid"))
+        } else {
+            Self::Seconds(Time::try_from(n).expect("n is valid"))
+        }
+    }
+}
+
+impl From<LockTime> for u32 {
+    fn from(n: LockTime) -> Self {
+        n.to_u32()
+    }
+}
+
+impl From<Height> for LockTime {
+    fn from(h: Height) -> Self {
+        LockTime::Blocks(h)
+    }
+}
+
+impl From<Time> for LockTime {
+    fn from(t: Time) -> Self {
+        LockTime::Seconds(t)
+    }
+}
+
+impl PartialOrd for LockTime {
+    fn partial_cmp(&self, other: &LockTime) -> Option<Ordering> {
+        use LockTime::*;
+
+        match (*self, *other) {
+            (Blocks(ref a), Blocks(ref b)) => a.partial_cmp(b),
+            (Seconds(ref a), Seconds(ref b)) => a.partial_cmp(b),
+            (_, _) => None,
+        }
+    }
+}
+
+impl fmt::Display for LockTime {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use LockTime::*;
+
+        if f.alternate() {
+            match *self {
+                Blocks(ref h) => write!(f, "block-height {}", h),
+                Seconds(ref t) => write!(f, "block-time {} (seconds since epoch)", t),
+            }
+        } else {
+            match *self {
+                Blocks(ref h) => write!(f, "{}", h),
+                Seconds(ref t) => write!(f, "{}", t),
+            }
+        }
+    }
+}
+
+impl Encodable for LockTime {
+    fn consensus_encode<W: io::Write>(&self, e: W) -> Result<usize, io::Error> {
+        let v = self.to_u32();
+        v.consensus_encode(e)
+    }
+}
+
+impl Decodable for LockTime {
+    #[inline]
+    fn consensus_decode<D: io::Read>(d: D) -> Result<Self, encode::Error> {
+        let v = u32::consensus_decode(d)?;
+        Ok(LockTime::from(v))
+    }
+}
+
+/// An absolute block height.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
+pub struct Height(u32); // Guaranteed to always contain a valid height value.
```

Comment:
Consider making "Guaranteed to always contain a valid height value." a port of docstring. Seems like a valuable thing to know.